db/migrations/20240930100434_add_snap_uuid_to_templates.up.sql

```diff
@@ -0,0 +1,11 @@
+BEGIN;
+
+ALTER TABLE templates_repository_configurations ADD COLUMN IF NOT EXISTS snapshot_uuid UUID DEFAULT NULL;
+
+ALTER TABLE templates_repository_configurations
+DROP CONSTRAINT IF EXISTS fk_templates_repository_configurations_snapshots,
+ADD CONSTRAINT fk_templates_repository_configurations_snapshots
+FOREIGN KEY (snapshot_uuid) REFERENCES snapshots(uuid)
+ON DELETE SET NULL;
```

Comment:
One question is whether it should restrict the deletion or set the column null.   If we restricted the deletion, it would require all snapshots to be updated for the template prior to deleting. 

If we allow for null, we could potentially end up with a null value.  What would be the ramifications of that?

Reply:
i think restricting the deletion instead makes sense, especially if we don't allow the `snapshot_uuid` to be null (see reply further down). and before deleting snapshots we should be updating the snapshots in the template anyway (looking at the [ticket](https://issues.redhat.com/browse/HMS-4742) for delete and bulk-delete)